diff --git a/sal/osl/unx/thread.cxx b/sal/osl/unx/thread.cxx
index b17f363511c6..0289a737ed26 100644
--- a/sal/osl/unx/thread.cxx
+++ b/sal/osl/unx/thread.cxx
@@ -51,7 +51,7 @@
 #include <osl/detail/android-bootstrap.h>
 #endif
 
-#if defined LINUX && ! defined __FreeBSD_kernel__
+#if defined LINUX && ! defined __FreeBSD_kernel__ && ! defined __EMSCRIPTEN__
 #include <sys/syscall.h>
 #endif
 
@@ -536,7 +536,7 @@ void SAL_CALL osl_yieldThread()
 void SAL_CALL osl_setThreadName(char const * name)
 {
     assert( name );
-#if defined LINUX && ! defined __FreeBSD_kernel__
+#if defined LINUX && ! defined __FreeBSD_kernel__ && ! defined __EMSCRIPTEN__
     const int LINUX_THREAD_NAME_MAXLEN = 15;
     if ( strlen( name ) > LINUX_THREAD_NAME_MAXLEN )
         SAL_INFO( "sal.osl", "osl_setThreadName truncated thread name to "
@@ -628,7 +628,7 @@ static oslThreadIdentifier insertThreadId (pthread_t hThread)
 
         pEntry->Handle = hThread;
 
-#if defined LINUX && ! defined __FreeBSD_kernel__
+#if defined LINUX && ! defined __FreeBSD_kernel__ && ! defined __EMSCRIPTEN__
 #if defined __GLIBC__ && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 30))
         // gettid returns a pid_t, which POSIX defines to be a signed integer type; assume that all
         // valid pid_t values on Linux are positive (zero is filtered out in the generic code
