diff --git a/basic/source/sbx/sbxconv.hxx b/basic/source/sbx/sbxconv.hxx
index 61a4a4788fd4..a1a4f835dd54 100644
--- a/basic/source/sbx/sbxconv.hxx
+++ b/basic/source/sbx/sbxconv.hxx
@@ -24,7 +24,10 @@
 #include <basic/sbx.hxx>
 #include <basic/sbxcore.hxx>
 #include <basic/sbxdef.hxx>
+#include <config_features.h>
+#if HAVE_FEATURE_SCRIPTING
 #include <runtime.hxx>
+#endif
 
 #include <o3tl/float_int_conversion.hxx>
 #include <o3tl/safeint.hxx>
@@ -35,10 +38,13 @@ class SbxArray;
 
 template <std::integral I> I ConvertWithOverflowTo(double f)
 {
-    // tdf#162711 - VBASupport mode requires banker’s rounding for integer conversions
+    // tdf#162711 - VBASupport mode requires banker's rounding for integer conversions
     f = rtl_math_round(f, 0,
+#if HAVE_FEATURE_SCRIPTING
                        SbiRuntime::isVBAEnabled() ? rtl_math_RoundingMode_HalfEven
-                                                  : rtl_math_RoundingMode_Corrected);
+                                                  :
+#endif
+                                                  rtl_math_RoundingMode_Corrected);
     if (!o3tl::convertsToAtMost(f, std::numeric_limits<I>::max()))
     {
         SbxBase::SetError(ERRCODE_BASIC_MATH_OVERFLOW);
